mysql-test/suite/tianmu/t/issue682.test

```diff
@@ -0,0 +1,57 @@
+--source include/have_tianmu.inc
+
+use test;
+set sql_mode='STRICT_TRANS_TABLES,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION';
+create table t1(a date not null);
```

Comment:
How about the not `date`  data type ? such as `datetime` or `int` or `timestamp`, etc.

Reply:
>Invalid DATE, DATETIME, or TIMESTAMP values are converted to the “zero” value of the appropriate type ('0000-00-00' or '0000-00-00 00:00:00'), if the SQL mode permits this conversion. The precise behavior depends on which if any of strict SQL mode and the [NO_ZERO_DATE](https://dev.mysql.com/doc/refman/8.0/en/sql-mode.html#sqlmode_no_zero_date) SQL mode are enabled; see [Section 5.1.11, “Server SQL Modes”](https://dev.mysql.com/doc/refman/8.0/en/sql-mode.html).

How about this?